W-15854291: Separate file commons

r-patole opened the pull request:
_Hi, thank you for your work. We understand that you want to merge your changes and move on from this issue, but we also want to maintain the safety, readability, and testability of our code. Because of this, we kindly ask that you confirm that you have fulfilled the prerequisites for each category of activity before merging your changes._

### If the Pull Request has a dependency update:

- [ ] I have read the Release Notes for the new version (and intermediate versions if the jump would include more than one). _Don't blindly trust that the dependency will honor SEMVER, always read carefully the release notes_
- [ ] Java 8+ support is maintained
- [ ] The new version has no vulnerabilities
- [ ] A team member reviewed the Pull Request.
- [ ] No backward compatibility is broken.
- [ ] If it is a "provided" dependency, it has been checked that if it is suggested, the version number is the same.

### For ALL the Releases:

- [ ] There is a task in GUS for this change.
- [ ] The corresponding Release Notes have been written and shared with the documentation team.
- [ ] The new code has tests and they have before and after methods to be sure that you are working with a clean environment and that you are leaving the environment clean after they finish.
- [ ] If you are using reflection, create a test to call the methods you are invoking, this way, if there is a change in the API, the test will be able to detect it.
- [ ] If the project has a parent pom and modules, the release pipeline only uploads the modules, so, please, make sure that the parent was deployed in the repositories before considering this release finished.
- [ ] Notify in the Slack Channel that the release is complete, so the Docs team can merge the Documentation and Release Notes PR. The docs team always checks in Anypoint Exchange that the connector is released before merging the release notes.
- [ ] Add the connector build to GUS, and assign that build to the GUS ticket.
- [ ] Create a change case request, based on the Change Case Management doc https://docs.google.com/document/d/1tMJlTTZLaXMLiYwxlMBFY1yJwBmejhc4-IP8HAXgDfY/edit#

**NOTE**: _(applies only for Core-Connectors connectors and modules) The release process ends when you merge the pull request that updates the support branch to the new snapshot, please don't forget to do this._

### Patch Version:

- [ ] The Pull Request has been peer-reviewed.
- [ ] No backward compatibility is broken.
- [ ] Documentation: (Required) Release Notes PR with compatibility table. Share it with the Docs team.

### Minor Version:

- [ ] Have a document with the specification of the release.
- [ ] Create a slack channel (eg: rl-conninfra-sftp-1.1.1-new-chiper) to coordinate the release of this version with the documentation team and the architects of our team (at least one week before the release). Share the spec doc with the docs team.
- [ ] The documentation fork has been done and has been updated with the changes related to the new functionalities.
- [ ] The pull request has been reviewed by a peer and the team leader.
- [ ] No backward compatibility is broken.
- [ ] Public Documentation: Minor releases nearly always require a revision of the documentation because the release usually includes new features as well as bug fixes. Even if there are no user-facing changes, documentation has to be versioned for every minor release.
  - [ ] (Required) Release Notes PR with compatibility table. Share it with the Docs team.
  - [ ] (Likely required) Reference guide. Run the script and share the file with the Docs team.
  - [ ] (Assess) Often a minor release impacts UI and examples, so the documentation (user guides, examples, troubleshooting guides) should be reviewed and updated accordingly. Notify the docs team if any updates are needed.

### Major Version:

- [ ] Have a document with the specification of the release.
- [ ] Create a slack channel (eg: rl-conninfra-sftp-2.0.0) to coordinate the release of this version with the documentation team and the architects of our team (at least one month before the release). Share the spec doc with the docs team.
- [ ] The documentation fork has been made and updated with the changes related to the new functionalities and possible compatibility problems with the previous version.
- [ ] The systems properties have been removed, detailing in the documentation the changes in the default behavior of the product.
- [ ] The TODO comments of the code have been reviewed.
- [ ] A colleague, the team leader, and the team architect have reviewed the Pull Request.
- [ ] Public Documentation: Major releases break backward compatibility and should never be released without documentation, as this breaks the user experience.
  - [ ] (Required) Upgrade and Migration Guide: Every major release must have a detailed upgrade guide (template) with information about what was changed and what steps the user has to take to ensure the connector works as expected. Share the details with the Docs team so they can update the guide.
  - [ ] (Required) Release Notes PR with compatibility table. Share it with the docs team.
  - [ ] (Required) Reference guide. Run the script and share the file with the Docs team.
  - [ ] User guide updates, new screenshots or examples (if applicable), share the updates with the docs team.

dipeshkumardutta-sfemu committed:
Remove file common dependency from source and test

dipeshkumardutta-sfemu committed:
Remove file common from test scope

dipeshkumardutta-sfemu committed:
Add privileged export

r-patole committed:
Upgraded min-version

r-patole committed:
Fixed File_Until and File_Since attribute

r-patole committed:
Ignoring Failing Java tests

r-patole committed:
Added javax.activation dependency

r-patole committed:
Added the MUnits for corresponding Java tests

r-patole committed:
Updated munit.version

r-patole committed:
Additional Munit tests

r-patole committed:
Read-write duplicate Test Case removed

r-patole committed:
Fixed failures due to different arguments

r-patole committed:
Formatting

r-patole committed:
Create directory with special name test fix

r-patole committed:
To-do: removed the OS assumption

r-patole committed:
To-do: ignored test case fixed

r-patole committed:
To-do: Remove redundant 'name' attribute

r-patole committed:
Fixed Unnecessary Stubbing Exception

chirag868 committed:
W-16004666: Ftp list operation change to return file path rather than the file content.

chirag868 committed:
Merge remote-tracking branch 'origin/feature/separate-file-commons-v2' into feature/separate-file-commons-v2

chirag868 committed:
W-16004666: Munit to list all files and validate that if it return file path rather than the file content.

chirag868 committed:
W-16004666: Updated java doc for list operation to return file path rather than the file content.

dipeshkumardutta-sfemu committed:
Update CODEOWNERS

dipeshkumardutta-sfemu committed:
Remove nexus team entry from CODEOWNERS file

chirag868 reviewed (commented):
!gate waiver add Sonar

nestor-diaz commented:
!gate waiver add Sonar

chirag868 reviewed (commented):
!gate waiver add Sonar

jonas19 commented:
!gate waiver add Sonar

garveshmehrotra-sfdc commented:
!gate waiver add Sonar

dipeshkumardutta-sfemu committed:
Update CODEOWNERS

chirag868 committed:
W-15854291: Versions update based on renovate bot PRs as below-
org.mule.module:mule-java-module to v1.2.13
io.fabric8:docker-maven-plugin to v0.44.0
org.apache.commons:commons-lang3 to v3.14.0
com.mycila:license-maven-plugin to v4.5
org.codehaus.mojo:build-helper-maven-plugin to v3.6.0
commons-net:commons-net to v3.9.0
slf4j monorepo to v2.0.13
commons-io:commons-io to v2.16.1
munit.version to v3.2.1

chirag868 committed:
Merge remote-tracking branch 'origin/feature/separate-file-commons-v2' into feature/separate-file-commons-v2

chirag868 committed:
W-15854291: For list operation, made changes to not read the file content anymore underneath it and return the filepath in response rather than the content

chirag868 committed:
W-16280695: Removed the params timeBetweenSizeCheck and timeBetweenSizeCheckUnit from the input and it's corresponding usage since we'll not read the file content anymore.

elcoya commented on src/main/java/org/mule/extension/ftp/internal/config/FileConnectorConfig.java:
```diff
@@ -0,0 +1,66 @@
+  @DefaultEncoding
+  private String muleEncoding;
+
+  /**
+   * Deprecated parameter
+   */
+  @Parameter
+  @Summary("This parameter is deprecated and will not be taken into account")
+  @DisplayName("Default write encoding(DEPRECATED)")
+  @Optional
+  @Placement(tab = ADVANCED_TAB)
+  private String defaultWriteEncoding;
```
If used anywhere, review. In any case, since it is deprecated, this needs to be removed.

chirag868 replied:
Included it as part of refactoring user story : https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001xkdp5YAA/view

elcoya replied:
Once more, removal of deprecated stuff need to happen for the new major, cannot be done afterwards.

elcoya reviewed (changes requested):
Wrapping it for now (40/123 files reviewed) so as to provide early feedback.
Please pay attention to the pom comment. Attribute `mule.version` is defined only in the parent pom.
On the other hand, there is a lot of classes in the api package that may do better just being a single class to avoid all the indirections, derived from the previous design (usage of file commons) that we are removing.
Finally, I can't avoid feeling that we should collapse all the hierarchy of FileSystem, AbstractFileSystem, AbstractExternalFileSystem right to the class FTPFileSystem. I understand this comes with a high risk, and can be deferred to a later change, but we must be very precise with the API defined now, so that all these changes won't break compatibility later on.

elcoya reviewed (commented):
Updated 61/123 files

chirag868 committed:
W-16280695: Moved constant file to internal package and removed mule runtime properties from pom since it's already included as part of a parent module pom

chirag868 commented on pom.xml:
```diff
@@ -19,35 +19,34 @@
     <description>A Mule connector that provides functionality to manage files in an FTP server</description>
 
     <properties>
-        <mule.version>4.1.2</mule.version>
```
agree, removed it.

chirag868 commented on src/main/java/org/mule/extension/ftp/api/AbstractFileAttributes.java:
```diff
@@ -0,0 +1,89 @@
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public String getName() {
+    return fileName;
+  }
```
Renaming or removal of **getName()** method is not a good idea here since we may end up with backward compatibility issues if customer is already using the method.
We faced similar issue in sftp connector as well where attribute used was **filename** but method was **getName()** , so we added **getFIlename()** to fix it considering the backward compatibility.

chirag868 replied:
done

chirag868 commented on src/main/java/org/mule/extension/ftp/api/ftp/FtpFileAttributes.java:
Created a user story : https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001xkdp5YAA/view for refactoring to be picked up later.

elcoya replied:
As mentioned in the previous comment. You can't remove classes from the connector `api` package without breaking compatibility. Will have to do it now.

chirag868 replied:
done

chirag868 commented on src/main/java/org/mule/extension/ftp/api/AbstractFileAttributes.java:
Created a user story : https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001xkdp5YAA/view for refactoring to be picked up later.

elcoya replied:
Cannot do that afterwards. `api` package is connector's API, and you cannot remove classes afterwards because that breaks compatibility. You will have to move it before the release.

chirag868 replied:
done

chirag868 committed:
W-16359329: Incorporating review comments by refactoring the code moved to ftp connector after eliminating dependency of mule-file-commons

chirag868 committed:
W-16359329: Fixed ftp-write-test-case.xml test case

chirag868 committed:
W-16359329: Removed the deprecated methods for read and write operations.

dipeshkumardutta-sfemu commented on src/main/java/org/mule/extension/ftp/api/ftp/FtpFileAttributes.java:
```diff
@@ -94,10 +105,6 @@ public void setSize(long size) {
 
-  /**
-   * {@inheritDoc}
-   */
```
Provide the comments in all the methods.

chirag868 replied:
added

chirag868 committed:
W-16359329: Added javadoc to FtpFileAttributes as per the review comments.

dipeshkumardutta-sfemu reviewed (dismissed):
LGTM

elcoya commented on src/main/java/org/mule/extension/ftp/api/ftp/FtpFileAttributes.java:
```diff
@@ -44,130 +44,196 @@ public class FtpFileAttributes extends AbstractFileAttributes {
-    super(createDefaultUri());
+    URI uri = createDefaultUri();
+    this.path = uri.getPath();
+    String name = FilenameUtils.getName(uri.getPath());
+    this.fileName = name != null ? name : "";
```
What's wrong with having the fileName `null` instead of empty `""`?

chirag868 replied:
This is an existing code which is moved while refactoring code from AbstractFileAttributes to FtpFileAttributes. Changing this null here will send value as null and may cause issue to customer.

chirag868 committed:
W-15854291: Refactored code to collapse hierarchy of FileSystem, AbstractFileSystem, AbstractExternalFileSystem, ExternalFileSystem and FtpFileSystem to  FileSystem and FtpFileSystem only.

elcoya commented on src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java:
```diff
@@ -0,0 +1,343 @@
+          initializePagingProvider(connection);
+        }
+        List<Result<String, FtpFileAttributes>> page = new LinkedList<>();
+        for (int i = 0; i < LIST_PAGE_SIZE && filesIterator.hasNext(); i++) {
```
Have you checked the instantiation of this iterator? We are paging the result, but internally, we first fetch the whole list of files. Seems like the paging here is just apparent, because now that the read of the file is gone, the paging only makes overhead.

chirag868 replied:
iterator is returning the results based on page size only.

elcoya commented on src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java:
```diff
@@ -0,0 +1,343 @@
+        filesIterator = files.iterator();
+      }
+
+      @Override
+      public java.util.Optional<Integer> getTotalResults(FileSystem connection) {
+        return java.util.Optional.of(files.size());
+      }
```
Doesn't seem to be the case (so no change is required here), but please be aware of the fact that if the result of this could change over the time, it could lead to the paging mechanism to hang waiting for results that hasn't come yet. in that case you could just make it return an Optional.empty() to avoid that situation.

chirag868 replied:
added check to return Optional.empty() in case files itself is null

elcoya commented on src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java:
```diff
@@ -0,0 +1,343 @@
+                                                           String directoryPath,
+                                                           boolean recursive,
+                                                           FileMatcher matchWith) {
+    fileSystem.changeToBaseDir();
```
What if the directoryPath is an absolute path? Why do this?

elcoya commented on src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java:
```diff
@@ -0,0 +1,343 @@
+      }
+
+      private void initializePagingProvider(FileSystem connection) {
+        connection.changeToBaseDir();
```
What if the directoryPath is an absolute path? Why do this?

chirag868 replied:
this is an existing call and internally checking if it's absolute path then it's not doing anything so should be good here.

elcoya commented on src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java:
```diff
@@ -0,0 +1,343 @@
+                                                          @Optional(defaultValue = "false") @Placement(
+                                                              tab = ADVANCED_TAB) boolean lock,
+                                                          Long timeBetweenSizeCheck) {
+    fileSystem.changeToBaseDir();
```
What if the path is an absolute path? Why do this?

chirag868 replied:
this is an existing call and internally checking if it's absolute path then it's not doing anything so should be good here.

elcoya commented on src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java:
```diff
@@ -0,0 +1,343 @@
+    }
+
+    validatePath(path, "path");
+    fileSystem.changeToBaseDir();
```
What if the path is an absolute path? Why do this?

chirag868 replied:
this is an existing call and internally checking if it's absolute path then it's not doing anything so should be good here.

elcoya commented on src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java:
```diff
@@ -0,0 +1,343 @@
+   */
+  protected void doCopy(FileConnectorConfig config, FileSystem fileSystem, String sourcePath,
+                        String targetPath, boolean createParentDirectories, boolean overwrite, String renameTo) {
+    fileSystem.changeToBaseDir();
```
Only applicable if the sourcePath and/or the targetPath are relative. Otherwise, it makes no sense to make this operation.

chirag868 replied:
this is an existing call and internally checking if it's absolute path then it's not doing anything so should be good here.

elcoya commented on src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java:
```diff
@@ -0,0 +1,343 @@
+   */
+  protected void doMove(FileConnectorConfig config, FileSystem fileSystem, String sourcePath,
+                        String targetPath, boolean createParentDirectories, boolean overwrite, String renameTo) {
+    fileSystem.changeToBaseDir();
```
What if the paths are absolute paths? Why do this?

chirag868 replied:
this is an existing call and internally checking if it's absolute path then it's not doing anything so should be good here.

elcoya commented on src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java:
```diff
@@ -0,0 +1,343 @@
+   * @throws IllegalArgumentException if {@code filePath} doesn't exist or is locked
+   */
+  protected void doDelete(FileSystem fileSystem, @Optional String path) {
+    fileSystem.changeToBaseDir();
```
again, absolute paths

chirag868 replied:
this is an existing call and internally checking if it's absolute path then it's not doing anything so should be good here.

elcoya commented on src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java:
```diff
@@ -0,0 +1,343 @@
+                                            to));
+    }
+
+    fileSystem.changeToBaseDir();
```
paths can be absolute

chirag868 replied:
this is an existing call and internally checking if it's absolute path then it's not doing anything so should be good here.

elcoya commented on src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java:
```diff
@@ -0,0 +1,343 @@
+
+  protected void doCreateDirectory(@Connection FileSystem fileSystem, String directoryPath) {
+    validatePath(directoryPath, "directory path");
+    fileSystem.changeToBaseDir();
```
paths can be absolute

chirag868 replied:
this is an existing call and internally checking if it's absolute path then it's not doing anything so should be good here.

elcoya commented on src/main/java/org/mule/extension/ftp/internal/operation/FileCommand.java:
```diff
@@ -0,0 +1,164 @@
+   */
+  protected abstract void doMkDirs(Path directoryPath);
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return whether the {@code Path} exists
+   */
+  protected void assureParentFolderExists(Path path, boolean createParentFolder) {
+    super.assureParentFolderExists(path, createParentFolder);
+  }
```
This implementation is redundant. Why is this here? Same for all following methods

chirag868 replied:
removed

elcoya commented on src/main/java/org/mule/extension/ftp/internal/source/FtpDirectoryListener.java:
```diff
@@ -74,7 +71,7 @@
 @Summary("Triggers when a new file is created in a directory")
 @Alias("listener")
 // TODO: MULE-13940 - add mimeType here too
```
Is this TODO still applicable??

elcoya reviewed (changes requested):
Left some inline comments, please review and fix. Still pending review some more files

elcoya commented on src/test/java/org/mule/extension/ftp/FtpCopyTestCase.java:
```diff
@@ -9,19 +9,22 @@
 import org.junit.Test;
 
 @Feature(FTP_EXTENSION)
+@Ignore
```
Why is this test ignored?

chirag868 replied:
@elcoya 
we are getting issue with these junits after we have upgraded parent pom version (mule-core-modules-parent) from **1.1.9 to 1.3.2**

**We have also written corresponding munits for the junits ignored:**
https://github.com/mulesoft/mule-ftp-connector/pull/426/commits/c38631d05efc0a23175e166e49aaa71a5f6e0ddb
https://github.com/mulesoft/mule-ftp-connector/pull/426/commits/4227899be936d81337a42db1cccd7cced7a9ae2b

**we are not sure how these junits can be fixed, please let us know if you know a way to fix these or can we remove these junits since we have corresponding munits for it?**

cc @dipeshkumardutta-sfemu

elcoya replied:
Seems like there are multiple issues with the docker images being used. Will have to troubleshoot and workout to solve them.

elcoya commented on src/test/java/org/mule/extension/ftp/FtpCreateDirectoryTestCase.java:
```diff
@@ -11,17 +11,19 @@
 import org.junit.Test;
 
 @Feature(FTP_EXTENSION)
+@Ignore
```
Same here... Ignore? why?!

elcoya commented on src/test/java/org/mule/extension/ftp/FtpDeleteTestCase.java:
```diff
@@ -8,14 +8,16 @@
 import org.junit.Test;
 
 @Feature(FTP_EXTENSION)
+@Ignore
```
One more. Why ignore?

elcoya commented on src/test/java/org/mule/extension/ftp/FtpDirectoryListenerFunctionalTestCase.java:
```diff
@@ -36,6 +36,7 @@
 import org.junit.Test;
 
 @Feature(FTP_EXTENSION)
+@Ignore
```
ignore

elcoya commented on src/test/java/org/mule/extension/ftp/FtpDirectoryListenerPollFailureConsistencyFunctionaTestCase.java:
```diff
@@ -26,6 +27,7 @@
 import io.qameta.allure.Description;
 import org.junit.Test;
 
+@Ignore
```
ignore...

elcoya reviewed (changes requested):
Team, please review the ignored tests. Why are they ignored? undo the ignore and make them work

chirag868 committed:
W-15854291: Refactored code to collapse following hierarchy:
1. AbstractFileCommand, ExternalFileCommand, FileCommand - FtpCommand
2. FileSystemProvider and FtpAbstractConnectionProvider -  FtpConnectionProvider

chirag868 committed:
W-15854291: Refactored code to collapse following hierarchy:
1. AbstractFileInputStreamSupplier and AbstractConnectedFileInputStreamSupplier - FtpFileInputStreamSupplier
2. AbstractPostActionGroup -  PostActionGroup
3. AbstractFtpCopyDelegate - FtpCopyDelegate

chirag868 committed:
W-15854291: Refactored code to collapse following hierarchy:
BaseFileSystemOperations - FtpOperations

chirag868 committed:
W-15854291: Incorporated review comments:
1. Make all the methods private which are used within that class
2. Remove implementing ConnectionProvider from FtpConnectionProvider
3. Changed default methods in FileSystem
4. Added descriptive javadoc to methods like getListCommand and similar

chirag868 committed:
W-15854291: Incorporated review comments:
keeping methods signature as same in api layer

chirag868 committed:
Merge pull request #434 from mulesoft/ftp-connector-major-refactoring

W-15854291: Code refactor to collapse hierarchies in major

chirag868 committed:
Merge remote-tracking branch 'origin/main' into feature/separate-file-commons-v2

# Conflicts:
#	.github/CODEOWNERS
#	pom.xml

chirag868 committed:
W-15854291: Incorporated review comments - removed lock(path) hierarchy from FileSystem and added null check for getTotalResults

chirag868 commented on src/main/java/org/mule/extension/ftp/internal/connection/AbstractExternalFileSystem.java:
```diff
@@ -0,0 +1,119 @@
+  protected final PathLock createLock(Path path) {
+    throw new UnsupportedOperationException("This method is not supported for an External File System. Use createLock(URI uri) instead.");
+  }
+
```
removed it.

chirag868 commented on src/main/java/org/mule/extension/ftp/internal/connection/AbstractExternalFileSystem.java:
removed the abstract class itself.

chirag868 commented on src/main/java/org/mule/extension/ftp/internal/connection/AbstractFileSystem.java:
```diff
@@ -0,0 +1,244 @@
+    acquireLock(lock);
+
+    return lock;
+  }
```
removed the whole hierarchy of lock(Path path) from FileSystem.

chirag868 commented on src/main/java/org/mule/extension/ftp/internal/connection/AbstractFileSystem.java:
```diff
@@ -0,0 +1,244 @@
+                                    format("Could not lock file '%s' because it's already owned by another process",
+                                           lock.getPath()));
+    }
+  }
```
removed

chirag868 commented on src/main/java/org/mule/extension/ftp/internal/connection/AbstractFileSystem.java:
```diff
@@ -0,0 +1,244 @@
+    if (isLocked(path)) {
+      throw new FileLockedException(format("File '%s' is locked by another process", path));
+    }
+  }
```
removed the whole hierarchy of lock(Path path).

chirag868 commented on src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java:
```diff
@@ -0,0 +1,343 @@
+ */
+public abstract class BaseFileSystemOperations {
+
+  private static final Integer LIST_PAGE_SIZE = 10;
```
since this is an existing size, keeping it same for now.

chirag868 committed:
W-15854291: Removed this TODO since it's valid for sftp and not for ftp connector

chirag868 committed:
W-15854291: As discussed changed page list size

chirag868 reviewed (commented):
As discussed, incremented the page list size to 10000 and created a story for future tasks:
https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001zu6kIYAQ/view

elcoya commented on src/main/java/org/mule/extension/ftp/internal/source/FtpDirectoryListener.java:
```diff
@@ -248,22 +243,22 @@ private FtpFileSystem openConnection() throws Exception {
 
-  private boolean processFile(Result<InputStream, FtpFileAttributes> file,
-                              PollContext<InputStream, FtpFileAttributes> pollContext) {
+  private boolean processFile(Result<String, FtpFileAttributes> file,
```
The path from the list operation can be turn into the InputStream inside this method, or it can be read before and sent into this method already as an InputStream as it was before.

chirag868 replied:
yes added the read call before and passing InputStream as it was before.

elcoya reviewed (changes requested):
@chirag868 & @dipeshkumardutta-sfemu Please review the comments on the DirectoryListener. Seems that you may have done that change unwittingly just following through with the change in the list operation. If you are changing that connector's Source, it should have been analyzed in the spec but there is absolutely no mention to it.

chirag868 committed:
W-15854291: Incorporated review comments-
1. FtpDirectoryListener : change directory source to InputStream which was there previously
2. FtpFileAttributes : keeping param path and name as consistent and its corresponding changes

chirag868 commented on src/main/java/org/mule/extension/ftp/api/ftp/FtpFileAttributes.java:
```diff
@@ -44,130 +44,196 @@ public class FtpFileAttributes extends AbstractFileAttributes {
+  protected final String path;
+
+  @Parameter
+  private final String fileName;
```
made them consistent as path and name.

chirag868 commented on src/main/java/org/mule/extension/ftp/internal/source/FtpDirectoryListener.java:
```diff
@@ -173,7 +169,7 @@ public void onTerminate(SourceCallbackContext ctx) {
 
   @Override
-  public void poll(PollContext<InputStream, FtpFileAttributes> pollContext) {
+  public void poll(PollContext<String, FtpFileAttributes> pollContext) {
```
unintentionally made this change, switched to InputStream now as previous.

chirag868 committed:
W-15854291: Fixed munits, renamed filename to name

chirag868 committed:
W-15854291: Changed FtpDirectoryListener approach to convert the path from the list operation into the InputStream before processFile method call

elcoya commented on src/main/java/org/mule/extension/ftp/internal/source/FtpDirectoryListener.java:
```diff
@@ -209,7 +212,12 @@ public void poll(PollContext<String, FtpFileAttributes> pollContext) {
         }
 
-        if (!processFile(file, pollContext)) {
+        InputStream contentStream = new ByteArrayInputStream(file.getOutput().getBytes(StandardCharsets.UTF_8));
```
You cannot use this class. It will load to memory (or at least try it) any file, regardless of its size. Instead, use the path of the file `file.getOutput()` to invoke whatever the `read` operation is doing.

chirag868 replied:
calling read operation now to get the content as Inputstream.

elcoya commented on src/main/java/org/mule/extension/ftp/internal/source/FtpDirectoryListener.java:
```diff
@@ -227,8 +235,8 @@ private void refreshMatcher() {
-  public void onRejectedItem(Result<String, FtpFileAttributes> result, SourceCallbackContext callbackContext) {
-    LOGGER.info("File processing was rejected for path: {}", result.getOutput());
+  public void onRejectedItem(Result<InputStream, FtpFileAttributes> result, SourceCallbackContext callbackContext) {
+    closeQuietly(result.getOutput());
```
If an item is rejected by the polling context you should at least inform it. If necessary, also close it.

chirag868 replied:
added.

elcoya reviewed (changes requested):
Need to fix the usage of ByteArrayInputStream for reading file contents. This class will put the whole file in memory. This will cause the memory to blow up during execution. Use, instead, whatever is using the read operation.

chirag868 committed:
W-15854291: Changed FtpDirectoryListener approach to make read call after the list operation to get the InputStream before processFile method call

arpitg-1 commented on src/main/java/org/mule/extension/ftp/internal/connection/FtpAbstractConnectionProvider.java:
```diff
@@ -1,350 +0,0 @@
- */
-@DisplayName("FTP Connection")
-@Summary("Connection to connect against an FTP server")
-public abstract class FtpAbstractConnectionProvider extends FileSystemProvider<FtpFileSystem> implements
```
We need this file for FTPS Connector. We'll face issues if we extend FtpConnectionProvider directly in ftps-connector.

chirag868 replied:
Reverted FileSystemProvider, FtpAbstractConnectionProvider and FtpConnectionProvider hierarchy to same as previous.

chirag868 committed:
W-15854291: Keeping the FileSystemProvider, FtpAbstractConnectionProvider and FtpConnectionProvider hierarchy same as previous since its required by FTPS Connector.

chirag868 committed:
W-15854291: Fixed ignored junits:
1. Downgraded core module parent to previous one.
2. Done refactoring by moving few files from internal to api package since it was not getting loaded by junit and was throwing NoClassDefFoundError.

chirag868 committed:
W-15854291: Fixed Ftp list test case

arpitg-1 commented on pom.xml:
```diff
@@ -19,35 +19,32 @@
     <description>A Mule connector that provides functionality to manage files in an FTP server</description>
 
     <properties>
```
@chirag868 I think we have reverted the parent pom version: mule-core-modules-parent. It should be 1.3.2

chirag868 replied:
upgraded to 1.3.2

arpitg-1 commented on pom.xml:
```diff
@@ -102,17 +92,15 @@
+            <version>${muleTestsRunnerVersion}</version>
             <scope>test</scope>
         </dependency>
 
```
I think we have missed an exclusion in mule-tests-infrastructure.

`<exclusions>
                <exclusion>
                    <artifactId>bcprov-jdk15on</artifactId>
                    <groupId>org.bouncycastle</groupId>
                </exclusion>
            </exclusions>`

chirag868 committed:
W-15854291: Core parent version upgrade to 1.3.2 and corresponding junits fix

chirag868 committed:
Merge branch 'main' into feature/separate-file-commons-v2

elcoya reviewed (approved):
looks good

dipeshkumardutta-sfemu reviewed (approved):
LGTM
